Source/WebGPU/WebGPU/Device.mm

```diff
@@ -254,6 +254,25 @@ static void captureFrame(id<MTLDevice> captureObject)
     }
 }
 
+void Device::generateAnOutOfMemoryError(String&& message)
+{
+    // https://gpuweb.github.io/gpuweb/#abstract-opdef-generate-an-out-of-memory-error
+
+    auto* scope = currentErrorScope(WGPUErrorFilter_OutOfMemory);
+
+    if (scope) {
+        if (!scope->error)
+            scope->error = Error { WGPUErrorType_OutOfMemory, WTFMove(message) };
+        return;
+    }
+
+    if (m_uncapturedErrorCallback) {
+        instance().scheduleWork([protectedThis = Ref { *this }, message = WTFMove(message)]() mutable {
```

Comment:
Hopefully message will be used on the same thread? (otherwise you need an isolatedCopy()).

Reply:
Correct it is the same thread. I don't think there is an obvious need to do this asynchronously as the operation is already async from the clients perspective, but there is some ambiguity in the specification and a previous discussion with @litherum reached the conclusion it should be async